### PR TITLE
Migration: Import user list filtering

### LINKED ISSUE #88502: Migration: Import user list filtering
Migration user import screen:

Add functionality to filter by search query and user roles.

Design: DASWajWXHUUjF4pu1yo4fI-fi-580%3A541

### PR DESCRIPTION
**Pull request by agrullon95**
Fixes https://github.com/Automattic/wp-calypso/issues/88502

## Proposed Changes

* Add functionality to filter by role or search query
CleanShot 2024-03-13 at 23 50 16@2x

## Testing Instructions

* Checkout branch or use live link.
* Navigate to `/sites?hosting-flow=true` and click `Migrate a site`.
* Go through the migration flow until you reach the import user screen.
* Verify you can filter by searching or user roles. Try to break it with different filter combinations.
* Verify the `select all users` toggle and invitations are still working.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

**Review by escapemanuele — APPROVED** [review]
Wow. This is impressive.

File: client/blocks/importer/wordpress/import-everything/import-users/index.tsx [hunk 1]
```diff
@@ -226,47 +359,41 @@ const ImportUsers = ( { site, onSubmit }: Props ) => {
...
-				userItem.checked = userExists ? userExists?.checked : true;
-
-				return userItem;
 			} );
+		}
 
-			const updatedCheckedUsersNumber = updatedUsersList?.filter( ( x ) => x.checked )?.length;
+		if ( userListFilters.selectedRoleFilters.length ) {
+			updatedUsersList = updatedUsersList.filter( ( user ) => {
+				const userRoles = user.user?.roles;
```
Comment (escapemanuele) [inline client/blocks/importer/wordpress/import-everything/import-users/index.tsx]: Are we able to use `getRole` here?

File: client/blocks/importer/wordpress/import-everything/import-users/index.tsx [hunk 2]
```diff
@@ -180,23 +210,126 @@ const ImportUsers = ( { site, onSubmit }: Props ) => {
 		}
 	};
 
+	const renderUsersListFilters = () => {
+		const searchUsers = ( query: string ) => {
+			setUserListFilters( { ...userListFilters, searchQuery: query } );
+		};
+
+		const roles = [
```
Comment (escapemanuele) [inline client/blocks/importer/wordpress/import-everything/import-users/index.tsx:218]: Can we reuse `getRoleBadgeText` ?

File: client/blocks/importer/wordpress/import-everything/import-users/index.tsx (hunk 1 above)
Comment (agrullon95) [reply client/blocks/importer/wordpress/import-everything/import-users/index.tsx]: Yes, `getRole` will work here. Updating this.

File: client/blocks/importer/wordpress/import-everything/import-users/index.tsx [hunk 3]
```diff
@@ -382,10 +382,7 @@ const ImportUsers = ( { site, onSubmit }: Props ) => {
 
 		if ( userListFilters.selectedRoleFilters.length ) {
 			updatedUsersList = updatedUsersList.filter( ( user ) => {
-				const userRoles = user.user?.roles;
-				return userRoles
-					? userRoles?.some( ( role ) => userListFilters.selectedRoleFilters.includes( role ) )
-					: false;
+				return userListFilters.selectedRoleFilters.includes( getRole( user.user ) );
```
Comment (escapemanuele) [inline client/blocks/importer/wordpress/import-everything/import-users/index.tsx:385]: 🥇

File: client/blocks/importer/wordpress/import-everything/import-users/index.tsx (hunk 2 above)
Comment (agrullon95) [reply client/blocks/importer/wordpress/import-everything/import-users/index.tsx:218]: I cant reuse `getRoleBadgeText`, but it provided more roles to add to the role filter. I will add those missing roles and also extract this `roles` array into its a constant variable inside `./utils` in a follow up.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/11604670

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @agrullon95 for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/11604670).

**Pull request merged**